subprojects/core/src/main/java/org/gradle/api/tasks/WriteProperties.java

```diff
@@ -191,30 +196,58 @@ public void setEncoding(String encoding) {
      * Returns the output file to write the properties to.
      */
     @OutputFile
+    @Optional
+    @Deprecated
     public File getOutputFile() {
-        return getServices().get(FileOperations.class).file(outputFile);
+        DeprecationLogger.deprecateProperty(getClass(), "outputFile").replaceWith("destinationFile")
+            .willBeRemovedInGradle9()
+            .withDslReference()
+            .nagUser();
+        return getDestinationFile().getAsFile().getOrNull();
     }
 
     /**
      * Sets the output file to write the properties to.
      *
      * @since 4.0
      */
+    @Deprecated
     public void setOutputFile(File outputFile) {
-        this.outputFile = outputFile;
+        DeprecationLogger.deprecateProperty(getClass(), "outputFile").replaceWith("destinationFile")
+            .willBeRemovedInGradle9()
+            .withDslReference()
+            .nagUser();
+        getDestinationFile().set(outputFile);
     }
 
     /**
      * Sets the output file to write the properties to.
      */
+    @Deprecated
     public void setOutputFile(Object outputFile) {
-        this.outputFile = outputFile;
+        DeprecationLogger.deprecateProperty(getClass(), "outputFile").replaceWith("destinationFile")
+            .willBeRemovedInGradle9()
+            .withDslReference()
+            .nagUser();
+
+        this.getDestinationFile().set(getServices().get(FileOperations.class).file(outputFile));
```

Comment:
:lipstick: unnecessary `this`
```suggestion
        getDestinationFile().set(getServices().get(FileOperations.class).file(outputFile));
```